includes/pipeline.php

```diff
@@ -70,24 +70,37 @@ public static function make_pipeline($resourceKey) {
         ]);
 
         $error = null;
-        try {
-            $cloud = new CloudRequestEngine(array("resourceKey" => $resourceKey));
-        }
-        catch (Exception $e) {
-            $error = $e->getMessage();
-            return array(
-                "pipeline" =>  null,
-                "available_engines" => null,
-                "error" => $error);
-        }
+        $cloud = new CloudRequestEngine(array("resourceKey" => $resourceKey));
         
+        $pipeline = Pipeline::buildPipelineWithPreprocessedEngine($builder, $cloud);
 
-        // Get engines available with the Resource Key
-        $engines = array_keys($cloud->flowElementProperties);
+        return array(
+            "pipeline" =>  $pipeline,
+            "available_engines" => array_keys($cloud->flowElementProperties),
+            "error" => $error);
+    }
+
+    private static function buildPipelineWithPreprocessedEngine($builder, $cloud)
+    {
+        // helper Pipeline
+        $pipeline = $builder->build();
+
+        // pre-process only the CloudRequestEngine element:
+        // ensure CloudRequestEngine is the first element in the pipeline
+        array_unshift($pipeline->flowElements, $cloud);
+        
+        // add CloudRequestEngine to keyed element list
+        $pipeline->flowElementsList[$cloud->dataKey] = $cloud;
+        
+        // process the element with helper FlowData
+        $flowData = $pipeline->createFlowData();
+        $flowData->pipeline->flowElementsList[$cloud->dataKey]->process($flowData);
```

Comment:
I think this rather defeats the purpose of making the evidenceKeys and cloudProperties properties lazy, because here you make them eager again, just through this workaround of processing...  

I think here we should have rather modified the test that expected a failure on construction to rather call process() on the constructed pipeline within the test and get a failure there.

Reply:
Updated.